lost/src/test/java/com/mapzen/android/lost/internal/LocationSettingsResultRequestTest.java

```diff
@@ -0,0 +1,606 @@
+package com.mapzen.android.lost.internal;
+
+import com.mapzen.android.lost.api.LocationRequest;
+import com.mapzen.android.lost.api.LocationSettingsRequest;
+import com.mapzen.android.lost.api.LocationSettingsResult;
+import com.mapzen.android.lost.api.ResultCallback;
+import com.mapzen.android.lost.api.Status;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import android.bluetooth.BluetoothAdapter;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.location.LocationManager;
+import android.support.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+public class LocationSettingsResultRequestTest {
+
+  Context context;
+  BluetoothAdapter bluetoothAdapter;
+  PackageManager pm;
+  LocationManager locationManager;
+  PendingIntentGenerator generator;
+  LocationSettingsRequest request;
+  LocationSettingsResultRequest resultRequest;
+
+  @Before
+  public void setup() {
+    context = Mockito.mock(Context.class);
+    bluetoothAdapter = Mockito.mock(BluetoothAdapter.class);
+    pm = Mockito.mock(PackageManager.class);
+    locationManager = Mockito.mock(LocationManager.class);
+    generator = new TestPendingIntentGenerator(context);
+
+    ArrayList<LocationRequest> requests = new ArrayList<>();
+    LocationRequest highAccuracy = LocationRequest.create().setPriority(
+        LocationRequest.PRIORITY_HIGH_ACCURACY); //gps + wifi
+    requests.add(highAccuracy);
+    request = new LocationSettingsRequest.Builder()
+        .addAllLocationRequests(requests)
+        .setNeedBle(true)
+        .build();
+
+    resultRequest = new LocationSettingsResultRequest(bluetoothAdapter, pm, locationManager,
+        generator, request);
+  }
+
+  @Test
+  public void await_shouldReturnSuccessfullResult() {
+    when(locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)).thenReturn(true);
+    when(pm.hasSystemFeature(PackageManager.FEATURE_LOCATION_GPS)).thenReturn(true);
+    when(locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER)).thenReturn(true);
+    when(pm.hasSystemFeature(PackageManager.FEATURE_LOCATION_NETWORK)).thenReturn(true);
+    when(bluetoothAdapter.isEnabled()).thenReturn(true);
```

Comment:
We should add `@SuppressWarnings("MissingPermission") ` to this class so Android Studio inspections stop complaining about permissions in tests.